tests/src/test/scala/cats/tests/MonadRecInstancesTests.scala

```diff
@@ -1,19 +1,52 @@
 package cats
 package tests
 
-import cats.data.{OptionT, StateT, Xor, XorT}
+import cats.data.{
```

Comment:
I agree, but would it be possible that we follow that up in a next PR since this PR and discussion is already getting pretty large? Refactoring the existing tests might be a good line to draw.